bundles/org.openhab.core.config.discovery.addon.ip/src/main/java/org/openhab/core/config/discovery/addon/ip/IpAddonFinder.java

```diff
@@ -96,34 +96,30 @@ public void deactivate() {
         stopScan();
     }
 
+    @Override
     public void setAddonCandidates(List<AddonInfo> candidates) {
         logger.debug("IpAddonFinder::setAddonCandidates({})", candidates.size());
         super.setAddonCandidates(candidates);
         startScan();
     }
 
-    synchronized void startScan() {
-        if (scanJob == null) {
-            scanJob = scheduler.schedule(this::scan, 1, TimeUnit.SECONDS);
```

Comment:
The `setAddonCandidates` method is called for each info provider.
In order to do the scan only once, but on the full set of candidates, we have to delay the execution.
At the same time we must make sure that a scheduled scan is rescheduled - or (after more than our delay) is executed once more.

Reply:
interesting....
I wonder if somebody could reach the setup wizard before the scan is done..... 20s might be too long.

Reply:
> 20s might be too long

I think it is a good compromise. I doubt that people rush through the wizard. Creating an account and choosing a good password should take them already a while. And if you consider that startup times on a RPi are around 5 minutes, I think we should be on the safe side.

Reply:
> btw: maybe we should add your explanation as comment to the code

Done.